File: Year-2023/LeetCode-April-Challenge/MinimizeMaximumofArray.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int minimizeArrayValue(vector<int>& nums) {
        long long sum = nums[0], avg = nums[0];
        for(int i = 1; i < nums.size(); ++i){
            sum += nums[i];
            if(nums[i] > avg){
                int x = sum / (i + 1);
                if(x >= avg){
                    if(sum % (i + 1)) avg = x + 1;
                    else avg = x;
                }
            }
        }
        return avg;
    }
};
int main(){
    vector<int> nums = {13,13,20,0,8,9,9};
    Solution *s = new Solution();
    cout<<s->minimizeArrayValue(nums)<<endl;
    return 0;
}

File: Year-2023/LeetCode-April-Challenge/MinimizeMaximumofArray.js
var minimizeArrayValue = function (nums) {
    let sum = nums[0], avg = nums[0];
    for (let i = 1; i < nums.length; ++i) {
        sum += nums[i];
        if (nums[i] > avg) {
            let x = Math.trunc(sum / (i + 1));
            if (x >= avg) {
                if (Math.trunc(sum % (i + 1))) avg = x + 1;
                else avg = x;
            }
        }
    }
    return avg;
};
let nums = [13, 13, 20, 0, 8, 9, 9];
console.log(minimizeArrayValue(nums));